### PR TITLE
REPLAY-1936 Session Replay Objective-C interface

### LINKED ISSUE #1387: Session Replay Beta - in objective - c
### The thing

Tell us the thing 🙂
Hi,

I am trying to check on Session Replay Beta branch. I am trying to switch swift code to objective-c (my AppDelegate in objective - c) , is there any help for objective - c code below?

```
import DatadogSessionReplay
var sessionReplay: SessionReplayController!

let configuration = SessionReplayConfiguration(privacy: .maskAll)
sessionReplay = SessionReplay.initialize(with: configuration)
sessionReplay.start()
```

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

Expose Session Replay to Objective-C.
Fix #1387

### How?

Add `SessionReplay` objective-c interface to `DatadogObjc` module.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [x] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [x] Run unit tests
- [ ] Run integration tests
- [x] Run smoke tests

### CODE REVIEW

File: DatadogObjc/Sources/SessionReplay/SessionReplay+objc.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    @objc
+    public static func enable(with configuration: DDSessionReplayConfiguration) {
+        SessionReplay.enable(with: configuration._swift)
+    }
+}
+
+/// Session Replay feature configuration.
+@objc
+public final class DDSessionReplayConfiguration: NSObject {
+    public var _swift: SessionReplay.Configuration = .init(replaySampleRate: 0)
```
Comment (ncreated) [inline DatadogObjc/Sources/SessionReplay/SessionReplay+objc.swift:33]: suggestion/ This we can hide, no?
```suggestion
    internal var _swift: SessionReplay.Configuration = .init(replaySampleRate: 0)
```

File: DatadogObjc/Sources/SessionReplay/SessionReplay+objc.swift [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    /// Record all content.
+    case allow
+
+    /// Mask all content.
+    case mask
+
+    /// Mask input elements, but record all other content.
+    case maskUserInput
+
+    public var _swift: SessionReplay.Configuration.PrivacyLevel {
```
Comment (ncreated) [inline DatadogObjc/Sources/SessionReplay/SessionReplay+objc.swift:91]: suggestion/
```suggestion
    internal var _swift: SessionReplay.Configuration.PrivacyLevel {
```

File: DatadogObjc/Sources/SessionReplay/SessionReplay+objc.swift [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        switch self {
+        case .allow: return .allow
+        case .mask: return .mask
+        case .maskUserInput: return .maskUserInput
+        default: return .mask
+        }
+    }
+
+    public init(_ swift: SessionReplay.Configuration.PrivacyLevel) {
```
Comment (ncreated) [inline DatadogObjc/Sources/SessionReplay/SessionReplay+objc.swift:100]: suggestion/
```suggestion
    internal init(_ swift: SessionReplay.Configuration.PrivacyLevel) {
```

**Review by ncreated — DISMISSED** [review]
🎯 💯 Looks good. Thanks for the earlier draft earlier - looking forward make Objc offering modular 🙂!

File: DatadogObjc/Sources/SessionReplay/SessionReplay+objc.swift (hunk 1 above)
Comment (maxep) [reply DatadogObjc/Sources/SessionReplay/SessionReplay+objc.swift:33]: Indeed, sorry about that!

**Review by ncreated — APPROVED** [review]

**Pull request merged**